include/bluetooth/bluetooth.h

```diff
@@ -45,6 +45,58 @@ extern "C" {
  */
 #define BT_ID_DEFAULT 0
 
+/** Opaque type representing a connection to a remote device */
+struct bt_adv;
+
+/* Don't require everyone to include conn.h */
+struct bt_conn;
+
+struct bt_le_adv_cb {
+	/** @brief The advertising set has finished sending adv data.
+	 *
+	 *  This callback notifies the application that the advertising set has
+	 *  finished sending advertising data.
+	 *  The advertising set can either have been stopped by a timeout or
+	 *  because because the specified number of advertising intervals has
+	 *  been reached.
+	 *
+	 *  @param adv The advertising set object.
+	 *  @param num_sent The number of advertising events completed.
+	 */
+	void (*sent)(struct bt_adv *adv, u8_t num_sent);
+
+	/** @brief The advertising set has accepted a new connection.
+	 *
+	 *  This callback notifies the application that the advertising set has
+	 *  accepted a new connection.
+	 *
+	 *  @param adv The advertising set object.
+	 *  @param num_sent The new connection object.
+	 */
+	void (*connected)(struct bt_adv *adv, struct bt_conn *conn);
+
+	/** @brief The advertising set has been scanned.
+	 *
+	 *  This callback notifies the application that the advertising set has
+	 *  been scanned by an active scanner.
+	 *
+	 *  @param adv The advertising set object.
+	 *  @param addr The Bluetooth device address of the active scanner.
+	 */
+	void (*scanned)(struct bt_adv *adv, bt_addr_le_t *addr);
```

Comment:
Maybe we could reuse this for legacy advertising, and we should make this future-proof. I suggest adding a parameter to both the `connected` and `scanned` methods:

```
struct bt_phy_params_t *props
```
which contains both channel and RSSI.

Reply:
I have added info structs for all callbacks. And the word `_ext_` is not mentioned in the advertiser callback register function. I think that should be future proof.

Reply:
This comment was made before we moved registering of advertising callbacks into the advertising set. I don't think it is worth it to revert that decision. So these will not be re-usable for the legacy advertising API. However normal advertising is still possible with the new advertising API. So this shouldn't be an issue. If the application wants to use the callbacks it just needs to switch to using the new advertising API.